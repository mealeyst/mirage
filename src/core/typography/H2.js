import PropTypes from 'prop-types'
import styled from 'styled-components'

/** @component */
const H2 = styled.h2`
    margin: 0;

    ${props => props.center ? 'text-align: center;' : ''}
    letter-spacing: .05rem;
    text-transform: ${props => props.lowercase
      ? 'inherit' : 'uppercase'};

    color: ${props => props.theme.colors.navy};

    font-family: ${props => props.theme.fonts.headerFont};
    font-size: ${props => props.fontSize};
    font-weight: 500;
    line-height: 1.0625;
`
H2.propTypes = {
  center: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string,
    PropTypes.object
  ]),
  theme: PropTypes.shape({
    fonts: PropTypes.shape({
      headerFont: PropTypes.string
    }),
    colors: PropTypes.shape({
      navy: PropTypes.string
    })
  }),
  lowercase: PropTypes.bool
}

H2.defaultProps = {
  fontSize: '3.2rem'
}

/** @component */
export default H2
