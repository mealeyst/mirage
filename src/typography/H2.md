```js
<H2>Oh yeah, good luck with that.</H2>
```

### H2 lowercase example:

```js
<H2 lowercase>{'I\'m a thing.'}</H2>
```
