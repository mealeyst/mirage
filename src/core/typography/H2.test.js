import React from 'react'
import 'jest-styled-components'

import H2 from './H2'

const { shallowWithTheme } = global

describe('(Styled Component) H2', () => {
  const createH2 = (props) => {
    return shallowWithTheme(<H2 {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createH2())
    .toMatchSnapshot()
  })

  test('setting lowercase prop', () => {
    expect(createH2({lowercase: true}))
    .toHaveStyleRule({
      'text-transform': 'inherit'
    })
  })

  test('setting center prop', () => {
    expect(createH2({center: true}))
    .toHaveStyleRule({
      'text-align': 'center'
    })
  })

  test('center prop not set', () => {
    expect(createH2({center: false}))
    .toHaveStyleRule({
      'text-align': undefined
    })
  })
})
