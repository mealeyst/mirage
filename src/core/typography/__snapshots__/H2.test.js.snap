// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) H2 matching the snapshot 1`] = `
.c0 {
  -webkit-letter-spacing: .05rem;
  -moz-letter-spacing: .05rem;
  -ms-letter-spacing: .05rem;
  letter-spacing: .05rem;
  text-transform: uppercase;
  color: #00003C;
  font-family: "din-cond","din-condensed-web",Arial,sans-serif;
  font-size: 3.2rem;
  font-weight: 500;
  line-height: 1.0625;
  margin: 2.6rem 0;
}

<h2
  className="c0"
  fontSize="3.2rem"
/>
`;
